File: code_wars/kyu_5/002-Simple_Pig_Latin.js
//###### Solution 1
function pigIt(str) {
  let result = [];
  const words = str.split(" "),
    pun = new Set([",", ".", "!", "?"]);

  for (let i = 0; i < words.length; i++) {
    if (pun.has(words[i][0])) {
      result.push(words[i]);
      continue;
    }
    result.push(words[i].slice(1) + words[i][0] + "ay");
  }

  return result.join(" ");
}

//###### Solution 1
function pigIt(str) {
  const pun = new Set([",", ".", "!", "?"]);
  return str
    .split(" ")
    .map((ele) => (pun.has(ele[0]) ? ele : `${ele.slice(1)}${ele[0]}ay`))
    .join(" ");
}

// ###### Problem Link: https://www.codewars.com/kata/513e08acc600c94f01000001/train/javascript

File: code_wars/kyu_5/002-Simple_Pig_Latin.py
###### Solution 1
def pig_it(text: str) -> str:
    result = []

    for word in text.split(" "):
        if word[0] in {"!", ".", ",", "?"}:
            result += word
            continue
        result.append(word[1:] + word[0] + "ay")

    return " ".join(result)


###### Solution 2
def pig_it(text: str) -> str:
    pun = {"!", ".", ",", "?"}
    return " ".join(
        [
            f"{word[1:]}{word[0]}ay" if word not in pun else word
            for word in text.split(" ")
        ]
    )


###### Problem Link: https://www.codewars.com/kata/520b9d2ad5c005041100000f/train/python
